sos/report/plugins/ext.py

```diff
@@ -0,0 +1,57 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+
+
+class Ext(Plugin, IndependentPlugin):
+    """This plugin collects information on mounted Ext2/3/4 filessystems on
+    the local system.
+
+    Users should expect `dumpe2fs -h` or `dumpe2fs` collections by this
+    plugin for each Ext2/3/4 filesystem that is recognized by lsblk.
+    """
+
+    short_desc = 'Ext2/3/4 filesystem'
+
+    plugin_name = 'ext'
+    profiles = ('storage',)
+    files = ('/sys/fs/ext4/', '/proc/fs/ext4/', '/proc/fs/jbd2/')
+
+    option_list = [
+        PluginOpt('dumpe2fs', default=False, desc='dump filesystem info'),
+        PluginOpt('frag', default=False,
+                  desc='collect filesystem fragmentation status')
+    ]
+
+    def setup(self):
+        dumpe2fs_opts = '-h'
+        if self.get_option('dumpe2fs'):
+            dumpe2fs_opts = ''
+        allfs = self.get_devices_by_fstype('ext')
+        if allfs:
+            for fs in allfs:
```

Comment:
The conditional is superfluous, if `allfs` is an empty list, python just continues on, skipping whatever is in the for-loop.

Reply:
Ok, but how to prevent not to collect  info from /proc/mount as fail safe? 
Something like this?
`
        for fs in allfs:
            self.add_cmd_output(f"xfs_info {fs}", tags="xfs_info")
                self.add_cmd_output(f"xfs_admin -l -u {fs}")

        if not allfs:
            mounts = '/proc/mounts'
            ext_fs_regex = r"^(/dev/.+).+xfs\s+"
            for dev in zip(self.do_regex_find_all(ext_fs_regex, mounts)):
                for e in dev:
                    parts = e.split()
                    self.add_cmd_output(f"xfs_info {parts[1]}",
                                        tags="xfs_info")
                    self.add_cmd_output(f"xfs_admin -l -u {parts[0]}")
`

Reply:
Are you suggesting something like:
```
    def setup(self):
        allfs = self.get_devices_by_fstype('xfs')
        for fs in allfs:
            self.add_cmd_output(f"xfs_info {fs}", tags="xfs_info")
                self.add_cmd_output(f"xfs_admin -l -u {fs}")

        if not allfs:
            mounts = '/proc/mounts'
            ext_fs_regex = r"^(/dev/.+).+xfs\s+"
            for dev in zip(self.do_regex_find_all(ext_fs_regex, mounts)):
                for e in dev:
                    parts = e.split()
                    self.add_cmd_output(f"xfs_info {parts[1]}",
                                        tags="xfs_info")
                    self.add_cmd_output(f"xfs_admin -l -u {parts[0]}")

        self.add_copy_spec(self.files)
```

Reply:
Yep, that's correct. We could even button it up a little more by `for fs in self.get_devices_by_fstype('xfs'):` provided there is nothing else using `allfs` later on.